Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -305,6 +332,25 @@ void HTMLInputElement::did_edit_text_node(Badge<BrowsingContext>)
     });
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#the-size-attribute
+unsigned HTMLInputElement::size() const
+{
+    auto size_attr = get_attribute(HTML::AttributeNames::size);
+    if (auto converted = size_attr.to_uint(); converted.has_value())
+        return *converted;
```

Comment:
> Can we ever end up in a situation where `size` is set to `"0"`? I know this is prevented in `parse_attribute`, but what if any other code sets it "directly" to 0?

This check is needed as otherwise it would reset the size attribute, if zero is set in javascript.
For example: 
```html
<input size="100">
<script>
    let inputElem = document.querySelector("input");
    inputElem.size = 0;
    console.log(inputElem.size); // Without the check this would log 20, with the check it logs 100
</script>
```

This happens because `set_attribute` will set the value of the attribute and then call `parse_attribute`.

Also the check in `parse_attribute` is needed to prevent `<input size="0">`.

Reply:
I'm not entirely sure how this is supposed to work, because Firefox instead raises an exception: `Uncaught DOMException: Index or size is negative or greater than the allowed amount` (I find this message hilarious because it claims that zero is "negative or greater than the allowed amount".)

Reply:
Yeah you're right. I now make it throw an `IndexSizeError` as that's the exception that firefox throws

Reply:
Do you find any documentation that says this is the right behavior? It'd be nice to have spec links instead of trying to guess.

Reply:
Found the [https://html.spec.whatwg.org/#limited-to-only-non-negative-numbers-greater-than-zero](spec) for it.
Changed the code to follow it properly!